GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -308,27 +300,30 @@ void InitCountArtificial(out ToolStripMenuItem toolStripGitStatus, out GitStatus
                         var status = e.State;
                         if (status == GitStatusMonitorState.Stopped)
                         {
-                            _toolStripGitStatus.Visible = false;
-                            _toolStripGitStatus.Text = "";
+                            toolStripButtonCommit.Enabled = false;
```

Comment:
Do we need to disable the button at any point? What can go wrong if it's clicked while the monitor's running?

Reply:
The button is disabled by this code if the Monitor has *stopped*.
I guess this should never happen. If I understand the code correctly, GE ended without any Commit button in this case before.
You're right the Commit button should still be clickable in this case.

@gerhardol wrote this, too:
> For normal repos the button should always be visible and active. 
> Commit Button must not be visible/enabled in bare repos. (Prefer disabled)

I prefer disabling, too.

> With this change, it may be possible to remove the requirement to restart to activate/deactivate commit count. But that is probably a follow up.

IIUC, then the GitStatusMonitor must always run or must be started and stopped accordingly.

Reply:
> IIUC, then the GitStatusMonitor must always run or must be started and stopped accordingly.

enabling/disbling should not be done here, only when opening a repo

Reply:
> Yes, and when settings have been changed. 

Then you are solving the "require restart" to enable/disable commit-count too.
(That was harder with the previous design.)

> Do you think it would be too expensive to let the GitStatusMonitor run all the time?

In a big repo, you may not want to run git-status periodically (even with performance improvements in 3.0, one core will be used for this 1/3 of the time when building; can be decreased by decreasing priorities but still load). To have the monitor active is not an issue.